sessionize/lib/src/commonMain/kotlin/co/touchlab/sessionize/platform/NotificationsModel.kt

```diff
@@ -157,11 +94,40 @@ object NotificationsModel {
 
     // Cancel Singular
 
-    private fun cancelReminderNotification(sessionid: Int) {
-        ServiceRegistry.notificationsApi.cancelLocalNotification(sessionid, notificationReminderTag)
-    }
-
     private fun cancelFeedbackNotification() {
         ServiceRegistry.notificationsApi.cancelLocalNotification(feedbackId, notificationFeedbackTag)
     }
+
+    fun recreateNotifications(){
+        backgroundTask({ sessionQueries.mySessions().executeAsList() }) { mySessions ->
+            ServiceRegistry.notificationsApi.cancelAllNotifications(mySessions)
```

Comment:
I'm not sure we want to cancel all notifications here, would it not cause existing notifications to re-alert even if they haven't changed? As you mentioned, though, I think you are looking into the right way to cancel.

Reply:
This has been changed now so that only the alarm is cancelled and re-created. This way if the notification has already been shown it won't re-alert.